# HG changeset patch
# User Andrew Cooper <andrew.cooper3@citrix.com>
# Date 1375959796 -3600
# Node ID 5dae5d46ab1d877b62383213867b5c2211ad5bc1
# Parent  7ff8434f9fcbf3fc93c6f6930f375be24093b34d
rombios/ata: Reading this status register has no relevant side effects

So taking two traps when one will do is pointless.  This removes 1 of 13
VMExits on the int 0x13 hotpath.

Signed-off-by: Andrew Cooper <andrew.cooper3@citrix.com>
Acked-by: Keir Fraser <keir@xen.org>

diff -r 7ff8434f9fcb -r 5dae5d46ab1d tools/firmware/rombios/rombios.c
--- a/tools/firmware/rombios/rombios.c
+++ b/tools/firmware/rombios/rombios.c
@@ -2540,7 +2540,6 @@ static int await_ide(when_done,base,time
   Bit32u time=0,last=0;
   Bit16u status;
   Bit8u result;
-  status = inb(base + ATA_CB_STAT); // for the times you're supposed to throw one away
   for(;;) {
     status = inb(base+ATA_CB_STAT);
     time++;
